### ISSUE TITLE
question  about performance on shapenet

### ISSUE THREAD

**Issue by summerice9** [Zhang-VISLab/Learning-to-Segment-3D-Point-Clouds-in-2D-Image-Space#5]
hi,

 I run your github code  on R2080 TI. The instance iou  on shapenet is only 0.75, which is much lower than that in your paper. I did not revise your code but directly ran your code. Is there anything need to be pay attention to ? For example, are there any  settings which may degrade the performance?

**Comment by YechengLyu** [comment]
Thanks for your comment. I have noticed that the performance varies between each attempts. Currently, I am still working on the code to make it faster and robust. After I finish the code and testing, I will update the repo together with my training log and weights.

Best,
Yecheng

**Comment by summerice9** [comment]
Hi Yecheng
Many thanks for your reply. In an another opened issue, you said you pre-generated the images and save them as a dataset. Could you tell we whether you do the data augmentation when you pre-generated the image? If I directly use your code to project each point cloud into an image without data augmentation, would the performance e.g.  iou be largely decreased ?

**Comment by YechengLyu** [comment]
I do data augmentation as well in my pre-generated image. The performance does not decrease much without data augmentation in the metrics of weighted accuracy, which is known as the point-wise accuracy. However, the iou may drop a lot, and that's why I need to update the code. 

As you may noticed, the IoU of each shape is calculated by averaging the IoU of each category. Hence, suppose we have a shape of a plane that contains a body, two wings and two engines, if we label all the points correctly but one to an extra category head, the iou ≈  Average([body,wing,engine,head])  = Average([1,1,1,0]) = 0.75.

**Comment by summerice9** [comment]
Hi Yecheng, you mean that your method can achieve high accuracy on testing images of  shapenet ?

I agree that a method achieves high accuracy but may not achieve high Iou.  shellnet and pointnet++ also provides evaluation codes for evaluating IoU on shapenet.  You may use their evaluation code to evaluate your results

**Comment by YechengLyu** [comment]
Thanks for your suggestion. I have brought the IoU evaluation code from the PointNet code and fit into my work in my first commit at "network_testing.py". With well training, I got both high accuracy and high IoU on test images for my CVPR paper. 
However, from the comments on Github I found that the training may not stable, and that why I am working on the updates for reproducible work.

**Comment by YechengLyu** [comment]
Hello I have released the code for training from pre-prepared dataset.

**Comment by summerice9** [comment]
Thank you very much.  I will download your code. 

> Hello I have released the code for training from pre-prepared dataset.

**Comment by ibadami** [comment]
> Hello I have released the code for training from pre-prepared dataset.

Thank you @YechengLyu . Does this mean that now the training is stable and archives almost consistent IOU as well?

**Comment by YechengLyu** [comment]
@ibadami It should be better, however I cannot guarantee it is good every time. You can take advantage of my reproduced models, which should be a good start.

**Comment by Zhang-VISLab** [comment]
Dear all, 

Is there still any issue with the code? Please do let us know if there is any, otherwise we will close this issue. Thanks.

**Comment by summerice9** [comment]
Hi Yecheng,

I have run your updated codes of training choice 2. The instance Iou is still 0.7523, which is inconsistent with your reported results i.e. 0.84.   I do not revise your code. Is there something I wrongly set ?

**Comment by YechengLyu** [comment]
Hello @summerice9 ,

Sorry to hear that. As I mentioned before, training using choice may lead to unstable performance, and that's why we released choice 1. You can try to pretrain the network using choice 1 and fine tune using choice 2, or you can simply fine tune using choice 2, which may help you get a well-trained model.

Yecheng

**Comment by Zhang-VISLab** [comment]
Dear all,

Thanks for your great comments. We are terribly sorry that we lost our CVPR 2020 code after submission. This repository is a reproduced work, and we released a pre-trained network model with 88.0% instance-mean-iou and 86.5% class-mean-iou. An updated ArXiv preprint is available. Could you please check our code to see whether there still exist some issues? If so, please open a new issue and we will resolve it asap. Thanks.
